### PR TITLE
Fix duplicate dataset creation

### LINKED ISSUE #66: Create dataset twice when changing dataset
The dataset is created twice with the same set of configs when changing the dataset, which makes the charts rendered twice with a blink.

It happens when changing dataset from A to B where configMeta of B is equal to or a subset of configMeta of A.

### PR DESCRIPTION
**Pull request by jcc37**
Resolve #66 

Tests for this bug should be added when resolving #54

### CODE REVIEW

File: src/services/data/data.service.ts [hunk 1]
```diff
@@ -52,15 +37,17 @@ export class DataService implements OnDestroy {
...
-            options: Object.keys(datasets),
+            defaultValue: defaultDatasetName,
+            options: datasetNames,
           },
           ...datasets[name].configMeta,
         };
 
         this.preferenceService.dataset$.next({
           ...createDefault(meta),
+          name,
```
Comment (Rhombus33) [inline src/services/data/data.service.ts:47]: Why does there need to be 2 properties with the same value?

File: src/services/data/data.service.ts [hunk 2]
```diff
@@ -52,15 +37,17 @@ export class DataService implements OnDestroy {
           },
           name: {
             type: 'select',
-            defaultValue: name,
-            options: Object.keys(datasets),
+            defaultValue: defaultDatasetName,
```
Comment (Rhombus33) [inline src/services/data/data.service.ts:39]: Why do we need to define the default for each one if it is always the same value?

Comment (jcc37) [reply src/services/data/data.service.ts:39]: We need to make new meta object contain `fields name` and `enabled`.  It is going to replace the meta object of previous dataset.

File: src/services/data/data.service.ts (hunk 1 above)
Comment (jcc37) [reply src/services/data/data.service.ts:47]: There are two subscription in `DataService`: One for updating meta and one for creating data when preference changes. When user changes the `name` in preference, the `_meta_name` does not change, so the subscription above will not be triggered to create new data. After `_meta_name` is changed, which means the meta in preference is update to date, that subscription will then be triggered.

https://github.com/googleinterns/guide-doge/pull/67/files/09a2e79376e508f458b5aa82e4446a43bd5a1c03#diff-04c7fa6e46d3da32b3b6976c27dd53edR19

**Pull request merged**